src/Surfnet/Stepup/Identity/Event/YubikeyPossessionProvenAndVerifiedEvent.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Identity\Event;
+
+use Surfnet\Stepup\DateTime\DateTime;
+use Surfnet\Stepup\Identity\AuditLog\Metadata;
+use Surfnet\Stepup\Identity\Value\CommonName;
+use Surfnet\Stepup\Identity\Value\Email;
+use Surfnet\Stepup\Identity\Value\IdentityId;
+use Surfnet\Stepup\Identity\Value\Institution;
+use Surfnet\Stepup\Identity\Value\SecondFactorId;
+use Surfnet\Stepup\Identity\Value\YubikeyPublicId;
+use Surfnet\StepupBundle\Value\SecondFactorType;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\SensitiveData\Forgettable;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\SensitiveData\SensitiveData;
+
+class YubikeyPossessionProvenAndVerifiedEvent extends IdentityEvent implements Forgettable
+{
+    /**
+     * @var \Surfnet\Stepup\Identity\Value\SecondFactorId
+     */
+    public $secondFactorId;
+
+    /**
+     * The Yubikey's public ID.
+     *
+     * @var \Surfnet\Stepup\Identity\Value\YubikeyPublicId
+     */
+    public $yubikeyPublicId;
+
+    /**
+     * @var \Surfnet\Stepup\Identity\Value\CommonName
+     */
+    public $commonName;
+
+    /**
+     * @var \Surfnet\Stepup\Identity\Value\Email
+     */
+    public $email;
+
+    /**
+     * @var \Surfnet\Stepup\DateTime\DateTime
+     */
+    public $registrationRequestedAt;
+
+    /**
+     * @var string
+     */
+    public $registrationCode;
+
+    /**
+     * @param IdentityId              $identityId
+     * @param Institution             $institution
+     * @param SecondFactorId          $secondFactorId
+     * @param YubikeyPublicId         $yubikeyPublicId
+     * @param CommonName              $commonName
+     * @param Email                   $email
+     * @param DateTime                $registrationRequestedAt
+     * @param string                  $registrationCode
+     */
+    public function __construct(
+        IdentityId $identityId,
+        Institution $institution,
+        SecondFactorId $secondFactorId,
+        YubikeyPublicId $yubikeyPublicId,
+        CommonName $commonName,
+        Email $email,
+        DateTime $registrationRequestedAt,
```

Comment:
`registrationRequestedAt` and `registrationCode` are not yet used? Probably due to the WIP status..

Reply:
This applies to other events

Reply:
I've checked and fixed all constructor assignments.